irohad/ordering/impl/ordering_gate_transport_grpc.cpp

```diff
@@ -29,24 +29,23 @@ grpc::Status OrderingGateTransportGrpc::onProposal(
     ::google::protobuf::Empty *response) {
   log_->info("receive proposal");
 
-  std::vector<shared_model::proto::Transaction> transactions;
-  for (const auto &tx : request->transactions()) {
-    transactions.emplace_back(tx);
-  }
-  log_->info("transactions in proposal: {}", transactions.size());
+  auto proposal_res = factory_->createProposal(*request);
+  proposal_res.match(
+      [this](
+          iroha::expected::Value<std::unique_ptr<shared_model::interface::Proposal>>
```

Comment:
Can be ```auto &v```, I assume

Reply:
Unfortunately variant cannot deduce type in this case